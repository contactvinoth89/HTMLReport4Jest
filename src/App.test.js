import '@testing-library/jest-dom';
import React from 'react';
import App from './App';
import { render, fireEvent } from '@testing-library/react';
const data = require('./data/nested.json');
const sampleData = require('./data/sample.json');
describe('Page Information', () => {
    test('Should contain minimal information', () => {
        window.resultData = data;
        const { container } = render(<App></App>);
        expect(
            container.querySelector('.infoWrapper').childNodes.length,
        ).toEqual(4);
    });
    test('Should contain project level information', () => {
        const infoData = { ...data };
        const reporterOptions = {
            information: [{ title: 'Title', value: 'Value', type: 'string' }],
        };
        infoData.reporterOptions = reporterOptions;
        window.resultData = infoData;
        const { container } = render(<App></App>);
        expect(
            container.querySelector('.infoWrapper').childNodes.length,
        ).toEqual(6);
    });
    test('Should contain project level information', () => {
        const infoData = { ...sampleData };
        window.resultData = infoData;
        const { container } = render(<App></App>);
        expect(
            container.querySelector('.infoWrapper').childNodes.length,
        ).toEqual(8);
    });
    test('Should contain open handle count', () => {
        const infoData = { ...sampleData };
        infoData.openHandles = ['handle'];
        window.resultData = infoData;
        const { container } = render(<App></App>);
        expect(
            container.querySelector('.infoWrapper').childNodes.length,
        ).toEqual(10);
    });
});

describe('Menu click', () => {
    test('Should show and hide tree on click', () => {
        window.resultData = data;
        const { container } = render(<App></App>);
        fireEvent.click(container.querySelector('#menu'));
        expect(
            container.querySelector('.sidenav').classList.contains('open'),
        ).toEqual(true);
        fireEvent.click(container.querySelector('#menu'));
        expect(
            container.querySelector('.sidenav').classList.contains('close'),
        ).toEqual(true);
    });
});

describe('Tree click', () => {
    test('Should call function on tree node click', () => {
        window.resultData = data;
        const { container } = render(<App></App>);
        fireEvent.click(container.querySelector('#menu'));
        expect(
            container.querySelector('.sidenav').classList.contains('open'),
        ).toEqual(true);
        const elem = container.querySelectorAll('.nodeTitle');
        fireEvent.click(elem[1]);
        expect(container.querySelector('.main')).toMatchSnapshot();
    });
});
