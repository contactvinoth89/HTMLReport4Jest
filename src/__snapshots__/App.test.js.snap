// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tree click Should call function on tree node click 1`] = `
<div
  class="main"
>
  <div
    class="summary"
  >
    <span
      class="dot green"
    >
      <a
        href="#Passed"
      >
        7
      </a>
      <span>
        Passed
      </span>
    </span>
    <span
      class="dot orange"
    >
      <a
        href="#Pending"
      >
        0
      </a>
      <span>
        Pending
      </span>
    </span>
    <span
      class="dot red"
    >
      <a
        href="#Failed"
      >
        2
      </a>
      <span>
        Failed
      </span>
    </span>
    <span
      class="dot blue"
    >
      <a
        href="#Total"
      >
        9
      </a>
      <span>
        Total
      </span>
    </span>
    <span
      class="dot gray"
    >
      <a
        href="#Todo"
      >
        0
      </a>
      <span>
        Todo
      </span>
    </span>
  </div>
  <div
    class="infoWrapper"
  >
    <div
      class="box box1"
    >
      Start Time
    </div>
    <div
      class="box box2"
    />
    <div
      class="box box1"
    >
      Interupted
    </div>
    <div
      class="box box2"
    >
      No
    </div>
  </div>
  <div>
    <p>
      Filter:
       
      <label
        class="checkboxLabel"
      >
        passed
        <input
          id="passed"
          type="checkbox"
          value="passed"
        />
        <span
          class="checkboxSpan"
        />
      </label>
      <label
        class="checkboxLabel"
      >
        failed
        <input
          id="failed"
          type="checkbox"
          value="failed"
        />
        <span
          class="checkboxSpan"
        />
      </label>
      <label
        class="checkboxLabel"
      >
        todo
        <input
          id="todo"
          type="checkbox"
          value="todo"
        />
        <span
          class="checkboxSpan"
        />
      </label>
      <label
        class="checkboxLabel"
      >
        pending
        <input
          id="pending"
          type="checkbox"
          value="pending"
        />
        <span
          class="checkboxSpan"
        />
      </label>
    </p>
  </div>
  <div
    class="row tableheader"
  >
    <div
      class="column testcase"
    >
      Test case
    </div>
    <div
      class="column result"
    >
      Result
    </div>
    <div
      class="column time"
    >
      Time
    </div>
    <div
      class="column information"
    >
      Additonal information
    </div>
  </div>
  <div
    class="tabs"
  >
    <input
      class="togglerCheckBox"
      id="elem_id4"
      type="checkbox"
    />
    <label
      class="tab-label"
      for="elem_id4"
    >
      outer 0
      <span
        class="numberCircleParent"
      >
        <span
          class="numberCircle green"
        >
          7
        </span>
        <span
          class="numberCircle orange"
        >
          0
        </span>
        <span
          class="numberCircle red"
        >
          2
        </span>
        <span
          class="numberCircle blue"
        >
          9
        </span>
        <span
          class="numberCircle gray"
        >
          0
        </span>
      </span>
    </label>
    <div
      class="tab-content row"
    >
      <div
        class="tabs"
      >
        <input
          class="togglerCheckBox"
          id="elem_id5"
          type="checkbox"
        />
        <label
          class="tab-label"
          for="elem_id5"
        >
          outer 1
          <span
            class="numberCircleParent"
          >
            <span
              class="numberCircle green"
            >
              7
            </span>
            <span
              class="numberCircle orange"
            >
              0
            </span>
            <span
              class="numberCircle red"
            >
              2
            </span>
            <span
              class="numberCircle blue"
            >
              9
            </span>
            <span
              class="numberCircle gray"
            >
              0
            </span>
          </span>
        </label>
        <div
          class="tab-content row"
        >
          <div
            class="tabs"
          >
            <input
              class="togglerCheckBox"
              id="elem_id6"
              type="checkbox"
            />
            <label
              class="tab-label"
              for="elem_id6"
            >
              outer 2
              <span
                class="numberCircleParent"
              >
                <span
                  class="numberCircle green"
                >
                  7
                </span>
                <span
                  class="numberCircle orange"
                >
                  0
                </span>
                <span
                  class="numberCircle red"
                >
                  2
                </span>
                <span
                  class="numberCircle blue"
                >
                  9
                </span>
                <span
                  class="numberCircle gray"
                >
                  0
                </span>
              </span>
            </label>
            <div
              class="tab-content row"
            >
              <div
                class="column testcase"
              >
                should pass
              </div>
              <div
                class="column result status"
              >
                <i>
                  <svg
                    aria-label="passed"
                    class="passed statusicon"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    version="1.1"
                    viewBox="0 0 16 16"
                    width="1em"
                  >
                    <path
                      d="M13.78 4.22a.75.75 0 010 1.06l-7.25 7.25a.75.75 0 01-1.06 0L2.22 9.28a.75.75 0 011.06-1.06L6 10.94l6.72-6.72a.75.75 0 011.06 0z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </i>
                passed
              </div>
              <div
                class="column time"
              >
                2 ms
              </div>
              <div
                class="column information"
              >
                <div
                  class="informationiconwrapper"
                >
                  <i>
                    <svg
                      aria-label="more information"
                      class="informationicon"
                      data-icon="exclamation-circle"
                      fill="currentColor"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm-32 232c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V296zm32 440a48.01 48.01 0 0 1 0-96 48.01 48.01 0 0 1 0 96z"
                      />
                    </svg>
                  </i>
                   
                  Info
                </div>
              </div>
              <div
                class="tabs"
              >
                <input
                  class="togglerCheckBox"
                  id="elem_id9"
                  type="checkbox"
                />
                <label
                  class="tab-label"
                  for="elem_id9"
                >
                  first outer 3
                  <span
                    class="numberCircleParent"
                  >
                    <span
                      class="numberCircle green"
                    >
                      1
                    </span>
                    <span
                      class="numberCircle orange"
                    >
                      0
                    </span>
                    <span
                      class="numberCircle red"
                    >
                      0
                    </span>
                    <span
                      class="numberCircle blue"
                    >
                      1
                    </span>
                    <span
                      class="numberCircle gray"
                    >
                      0
                    </span>
                  </span>
                </label>
                <div
                  class="tab-content row"
                >
                  <div
                    class="tabs"
                  >
                    <input
                      class="togglerCheckBox"
                      id="elem_id10"
                      type="checkbox"
                    />
                    <label
                      class="tab-label"
                      for="elem_id10"
                    >
                      outer 4
                      <span
                        class="numberCircleParent"
                      >
                        <span
                          class="numberCircle green"
                        >
                          1
                        </span>
                        <span
                          class="numberCircle orange"
                        >
                          0
                        </span>
                        <span
                          class="numberCircle red"
                        >
                          0
                        </span>
                        <span
                          class="numberCircle blue"
                        >
                          1
                        </span>
                        <span
                          class="numberCircle gray"
                        >
                          0
                        </span>
                      </span>
                    </label>
                    <div
                      class="tab-content row"
                    >
                      <div
                        class="column testcase"
                      >
                        should pass
                      </div>
                      <div
                        class="column result status"
                      >
                        <i>
                          <svg
                            aria-label="passed"
                            class="passed statusicon"
                            fill="currentColor"
                            height="1em"
                            role="img"
                            version="1.1"
                            viewBox="0 0 16 16"
                            width="1em"
                          >
                            <path
                              d="M13.78 4.22a.75.75 0 010 1.06l-7.25 7.25a.75.75 0 01-1.06 0L2.22 9.28a.75.75 0 011.06-1.06L6 10.94l6.72-6.72a.75.75 0 011.06 0z"
                              fill-rule="evenodd"
                            />
                          </svg>
                        </i>
                        passed
                      </div>
                      <div
                        class="column time"
                      />
                      <div
                        class="column information"
                      >
                        <div
                          class="informationiconwrapper"
                        >
                          <i>
                            <svg
                              aria-label="more information"
                              class="informationicon"
                              data-icon="exclamation-circle"
                              fill="currentColor"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm-32 232c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V296zm32 440a48.01 48.01 0 0 1 0-96 48.01 48.01 0 0 1 0 96z"
                              />
                            </svg>
                          </i>
                           
                          Info
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="tabs"
              >
                <input
                  class="togglerCheckBox"
                  id="elem_id13"
                  type="checkbox"
                />
                <label
                  class="tab-label"
                  for="elem_id13"
                >
                  outer 3
                  <span
                    class="numberCircleParent"
                  >
                    <span
                      class="numberCircle green"
                    >
                      1
                    </span>
                    <span
                      class="numberCircle orange"
                    >
                      0
                    </span>
                    <span
                      class="numberCircle red"
                    >
                      1
                    </span>
                    <span
                      class="numberCircle blue"
                    >
                      2
                    </span>
                    <span
                      class="numberCircle gray"
                    >
                      0
                    </span>
                  </span>
                </label>
                <div
                  class="tab-content row"
                >
                  <div
                    class="tabs"
                  >
                    <input
                      class="togglerCheckBox"
                      id="elem_id14"
                      type="checkbox"
                    />
                    <label
                      class="tab-label"
                      for="elem_id14"
                    >
                      outer 4
                      <span
                        class="numberCircleParent"
                      >
                        <span
                          class="numberCircle green"
                        >
                          1
                        </span>
                        <span
                          class="numberCircle orange"
                        >
                          0
                        </span>
                        <span
                          class="numberCircle red"
                        >
                          1
                        </span>
                        <span
                          class="numberCircle blue"
                        >
                          2
                        </span>
                        <span
                          class="numberCircle gray"
                        >
                          0
                        </span>
                      </span>
                    </label>
                    <div
                      class="tab-content row"
                    >
                      <div
                        class="column testcase"
                      >
                        should pass
                      </div>
                      <div
                        class="column result status"
                      >
                        <i>
                          <svg
                            aria-label="passed"
                            class="passed statusicon"
                            fill="currentColor"
                            height="1em"
                            role="img"
                            version="1.1"
                            viewBox="0 0 16 16"
                            width="1em"
                          >
                            <path
                              d="M13.78 4.22a.75.75 0 010 1.06l-7.25 7.25a.75.75 0 01-1.06 0L2.22 9.28a.75.75 0 011.06-1.06L6 10.94l6.72-6.72a.75.75 0 011.06 0z"
                              fill-rule="evenodd"
                            />
                          </svg>
                        </i>
                        passed
                      </div>
                      <div
                        class="column time"
                      />
                      <div
                        class="column information"
                      >
                        <div
                          class="informationiconwrapper"
                        >
                          <i>
                            <svg
                              aria-label="more information"
                              class="informationicon"
                              data-icon="exclamation-circle"
                              fill="currentColor"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm-32 232c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V296zm32 440a48.01 48.01 0 0 1 0-96 48.01 48.01 0 0 1 0 96z"
                              />
                            </svg>
                          </i>
                           
                          Info
                        </div>
                      </div>
                      <div
                        class="column testcase"
                      >
                        should fail
                      </div>
                      <div
                        class="column result status"
                      >
                        <i>
                          <svg
                            aria-label="failed"
                            class="failed statusicon"
                            fill="currentColor"
                            height="1em"
                            role="img"
                            version="1.1"
                            viewBox="0 0 16 16"
                            width="1em"
                          >
                            <path
                              d="M3.72 3.72a.75.75 0 011.06 0L8 6.94l3.22-3.22a.75.75 0 111.06 1.06L9.06 8l3.22 3.22a.75.75 0 11-1.06 1.06L8 9.06l-3.22 3.22a.75.75 0 01-1.06-1.06L6.94 8 3.72 4.78a.75.75 0 010-1.06z"
                              fill-rule="evenodd"
                            />
                          </svg>
                        </i>
                        failed
                      </div>
                      <div
                        class="column time"
                      >
                        110 ms
                      </div>
                      <div
                        class="column information"
                      >
                        <div
                          class="informationiconwrapper"
                        >
                          <i>
                            <svg
                              aria-label="more information"
                              class="informationicon"
                              data-icon="exclamation-circle"
                              fill="currentColor"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm-32 232c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V296zm32 440a48.01 48.01 0 0 1 0-96 48.01 48.01 0 0 1 0 96z"
                              />
                            </svg>
                          </i>
                           
                          Info
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="tabs"
              >
                <input
                  class="togglerCheckBox"
                  id="elem_id19"
                  type="checkbox"
                />
                <label
                  class="tab-label"
                  for="elem_id19"
                >
                  next outer 3
                  <span
                    class="numberCircleParent"
                  >
                    <span
                      class="numberCircle green"
                    >
                      3
                    </span>
                    <span
                      class="numberCircle orange"
                    >
                      0
                    </span>
                    <span
                      class="numberCircle red"
                    >
                      1
                    </span>
                    <span
                      class="numberCircle blue"
                    >
                      4
                    </span>
                    <span
                      class="numberCircle gray"
                    >
                      0
                    </span>
                  </span>
                </label>
                <div
                  class="tab-content row"
                >
                  <div
                    class="column testcase"
                  >
                    should be around
                  </div>
                  <div
                    class="column result status"
                  >
                    <i>
                      <svg
                        aria-label="passed"
                        class="passed statusicon"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        version="1.1"
                        viewBox="0 0 16 16"
                        width="1em"
                      >
                        <path
                          d="M13.78 4.22a.75.75 0 010 1.06l-7.25 7.25a.75.75 0 01-1.06 0L2.22 9.28a.75.75 0 011.06-1.06L6 10.94l6.72-6.72a.75.75 0 011.06 0z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </i>
                    passed
                  </div>
                  <div
                    class="column time"
                  />
                  <div
                    class="column information"
                  >
                    <div
                      class="informationiconwrapper"
                    >
                      <i>
                        <svg
                          aria-label="more information"
                          class="informationicon"
                          data-icon="exclamation-circle"
                          fill="currentColor"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm-32 232c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V296zm32 440a48.01 48.01 0 0 1 0-96 48.01 48.01 0 0 1 0 96z"
                          />
                        </svg>
                      </i>
                       
                      Info
                    </div>
                  </div>
                  <div
                    class="tabs"
                  >
                    <input
                      class="togglerCheckBox"
                      id="elem_id22"
                      type="checkbox"
                    />
                    <label
                      class="tab-label"
                      for="elem_id22"
                    >
                      outer 4
                      <span
                        class="numberCircleParent"
                      >
                        <span
                          class="numberCircle green"
                        >
                          1
                        </span>
                        <span
                          class="numberCircle orange"
                        >
                          0
                        </span>
                        <span
                          class="numberCircle red"
                        >
                          1
                        </span>
                        <span
                          class="numberCircle blue"
                        >
                          2
                        </span>
                        <span
                          class="numberCircle gray"
                        >
                          0
                        </span>
                      </span>
                    </label>
                    <div
                      class="tab-content row"
                    >
                      <div
                        class="column testcase"
                      >
                        should pass
                      </div>
                      <div
                        class="column result status"
                      >
                        <i>
                          <svg
                            aria-label="passed"
                            class="passed statusicon"
                            fill="currentColor"
                            height="1em"
                            role="img"
                            version="1.1"
                            viewBox="0 0 16 16"
                            width="1em"
                          >
                            <path
                              d="M13.78 4.22a.75.75 0 010 1.06l-7.25 7.25a.75.75 0 01-1.06 0L2.22 9.28a.75.75 0 011.06-1.06L6 10.94l6.72-6.72a.75.75 0 011.06 0z"
                              fill-rule="evenodd"
                            />
                          </svg>
                        </i>
                        passed
                      </div>
                      <div
                        class="column time"
                      />
                      <div
                        class="column information"
                      >
                        <div
                          class="informationiconwrapper"
                        >
                          <i>
                            <svg
                              aria-label="more information"
                              class="informationicon"
                              data-icon="exclamation-circle"
                              fill="currentColor"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm-32 232c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V296zm32 440a48.01 48.01 0 0 1 0-96 48.01 48.01 0 0 1 0 96z"
                              />
                            </svg>
                          </i>
                           
                          Info
                        </div>
                      </div>
                      <div
                        class="column testcase"
                      >
                        should fail
                      </div>
                      <div
                        class="column result status"
                      >
                        <i>
                          <svg
                            aria-label="failed"
                            class="failed statusicon"
                            fill="currentColor"
                            height="1em"
                            role="img"
                            version="1.1"
                            viewBox="0 0 16 16"
                            width="1em"
                          >
                            <path
                              d="M3.72 3.72a.75.75 0 011.06 0L8 6.94l3.22-3.22a.75.75 0 111.06 1.06L9.06 8l3.22 3.22a.75.75 0 11-1.06 1.06L8 9.06l-3.22 3.22a.75.75 0 01-1.06-1.06L6.94 8 3.72 4.78a.75.75 0 010-1.06z"
                              fill-rule="evenodd"
                            />
                          </svg>
                        </i>
                        failed
                      </div>
                      <div
                        class="column time"
                      />
                      <div
                        class="column information"
                      >
                        <div
                          class="informationiconwrapper"
                        >
                          <i>
                            <svg
                              aria-label="more information"
                              class="informationicon"
                              data-icon="exclamation-circle"
                              fill="currentColor"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm-32 232c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V296zm32 440a48.01 48.01 0 0 1 0-96 48.01 48.01 0 0 1 0 96z"
                              />
                            </svg>
                          </i>
                           
                          Info
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="column testcase"
                  >
                    should be around too
                  </div>
                  <div
                    class="column result status"
                  >
                    <i>
                      <svg
                        aria-label="passed"
                        class="passed statusicon"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        version="1.1"
                        viewBox="0 0 16 16"
                        width="1em"
                      >
                        <path
                          d="M13.78 4.22a.75.75 0 010 1.06l-7.25 7.25a.75.75 0 01-1.06 0L2.22 9.28a.75.75 0 011.06-1.06L6 10.94l6.72-6.72a.75.75 0 011.06 0z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </i>
                    passed
                  </div>
                  <div
                    class="column time"
                  />
                  <div
                    class="column information"
                  >
                    <div
                      class="informationiconwrapper"
                    >
                      <i>
                        <svg
                          aria-label="more information"
                          class="informationicon"
                          data-icon="exclamation-circle"
                          fill="currentColor"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm-32 232c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V296zm32 440a48.01 48.01 0 0 1 0-96 48.01 48.01 0 0 1 0 96z"
                          />
                        </svg>
                      </i>
                       
                      Info
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="tabs"
              >
                <input
                  class="togglerCheckBox"
                  id="elem_id29"
                  type="checkbox"
                />
                <label
                  class="tab-label"
                  for="elem_id29"
                >
                  last outer 3
                  <span
                    class="numberCircleParent"
                  >
                    <span
                      class="numberCircle green"
                    >
                      1
                    </span>
                    <span
                      class="numberCircle orange"
                    >
                      0
                    </span>
                    <span
                      class="numberCircle red"
                    >
                      0
                    </span>
                    <span
                      class="numberCircle blue"
                    >
                      1
                    </span>
                    <span
                      class="numberCircle gray"
                    >
                      0
                    </span>
                  </span>
                </label>
                <div
                  class="tab-content row"
                >
                  <div
                    class="tabs"
                  >
                    <input
                      class="togglerCheckBox"
                      id="elem_id30"
                      type="checkbox"
                    />
                    <label
                      class="tab-label"
                      for="elem_id30"
                    >
                      outer 4
                      <span
                        class="numberCircleParent"
                      >
                        <span
                          class="numberCircle green"
                        >
                          1
                        </span>
                        <span
                          class="numberCircle orange"
                        >
                          0
                        </span>
                        <span
                          class="numberCircle red"
                        >
                          0
                        </span>
                        <span
                          class="numberCircle blue"
                        >
                          1
                        </span>
                        <span
                          class="numberCircle gray"
                        >
                          0
                        </span>
                      </span>
                    </label>
                    <div
                      class="tab-content row"
                    >
                      <div
                        class="column testcase"
                      >
                        should pass
                      </div>
                      <div
                        class="column result status"
                      >
                        <i>
                          <svg
                            aria-label="passed"
                            class="passed statusicon"
                            fill="currentColor"
                            height="1em"
                            role="img"
                            version="1.1"
                            viewBox="0 0 16 16"
                            width="1em"
                          >
                            <path
                              d="M13.78 4.22a.75.75 0 010 1.06l-7.25 7.25a.75.75 0 01-1.06 0L2.22 9.28a.75.75 0 011.06-1.06L6 10.94l6.72-6.72a.75.75 0 011.06 0z"
                              fill-rule="evenodd"
                            />
                          </svg>
                        </i>
                        passed
                      </div>
                      <div
                        class="column time"
                      />
                      <div
                        class="column information"
                      >
                        <div
                          class="informationiconwrapper"
                        >
                          <i>
                            <svg
                              aria-label="more information"
                              class="informationicon"
                              data-icon="exclamation-circle"
                              fill="currentColor"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm-32 232c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V296zm32 440a48.01 48.01 0 0 1 0-96 48.01 48.01 0 0 1 0 96z"
                              />
                            </svg>
                          </i>
                           
                          Info
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
